### DISCUSSION TITLE
Regional Prompter error in Img2img

### DISCUSSION THREAD

**Discussion by Zpno** [vladmandic/sdnext#2274] [Q&A] [answered]
Using the versions of SD Next `c6226605` and Regional Prompter `029ecedd`, I have no problems generating pictures in txt2img but as soon as I use it in img2img I get this : 

>ERROR    Running script process: extensions\sd-webui-regional-prompter\scripts\rp.py: AttributeError
╭───────────────────────────────────────── Traceback (most recent call last) ──────────────────────────────────────────╮
│ E:\SD Next\modules\scripts.py:470 in process                                                                         │
│                                                                                                                      │
│   469 │   │   │   │   args = p.per_script_args.get(script.title(), p.script_args[script.args_f                       │
│ ❱ 470 │   │   │   │   script.process(p, *args, **kwargs)                                                             │
│   471 │   │   │   except Exception as e:                                                                             │
│                                                                                                                      │
│ E:\SD Next\extensions\sd-webui-regional-prompter\scripts\rp.py:431 in process                                        │
│                                                                                                                      │
│    430 │   │   │   self.hr = p.enable_hr                                                                             │
│ ❱  431 │   │   │   self.hr_w = (p.hr_resize_x if p.hr_resize_x > p.width else p.width * p.hr_sc                      │
│    432 │   │   │   self.hr_h = (p.hr_resize_y if p.hr_resize_y > p.height else p.height * p.hr_                      │
╰──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────╯
AttributeError: 'StableDiffusionProcessingImg2Img' object has no attribute 'hr_resize_x'

I still get images but the prompts are polluting each other, making Regional Prompter useless.

I'm on Windows 11, with Python 3.10.11. Tried with a fresh install of SD Next, still the same error. But it works fine with a fresh install of Automatic1111.

Anyone is experiencing the same ? Is this an issue due to SD Next or Regional Prompter ?

**Comment by lbeltrame** [comment]
Looks like an issue with Regional Prompter, as far as I can see. You might want to open an issue there.

**Comment by vladmandic** [comment, marked as answer]
its because img2img does NOT have those properties - it note that `hr_resize` refers to hires - and img2img does not have hires pass.
so its extension's wrong assumption that those properties would always be present - just wrong.
if a1111 sets those params in img2img but doesn't use them at all, does that make it any better? imo, it doesn't, it just hides an underlying problem in the extension.
